### PR TITLE
machines: Storage pool creation fixes

### PR DESCRIPTION
**Pull request by KKoukiou**

### CODE REVIEW

**Comment by martinpitt** [comment]
[this test failure](https://209.132.184.41:8493/logs/pull-11873-20190520-154848-f0f5d623-verify-rhel-8-0-distropkg/log.html#260) looks relevant, not sure if it's related to this PR or a flake, though:
```
TypeError: Method Promise.prototype.then called on incompatible receiver #<Object>
```

File: test/verify/check-machines-dbus [hunk 1]
```diff
@@ -812,6 +812,11 @@ class TestMachinesDBus(machineslib.TestMachines):
                 b.click(".modal-footer button:contains(Create)")
 
                 if (not self.xfail):
+                    # For the pool types that take some seconds to get created check that the spinner appears.
+                    # Other pool types have the spinner as well, but let's not check this in the tests to avoid race conditions,
+                    # where the spinner disappears very fast
+                    if self.pool_type  in ["netfs", "iscsi", "iscsi-direct"]:
+                        b.wait_present(".modal-footer div.spinner")
```
Comment (martinpitt) [inline test/verify/check-machines-dbus]: You could also wait for the disabled button here? (which is the more important of the two widgets)

**Comment by martinpitt** [comment]
Yep, the [rhel-8-0 failure](https://209.132.184.41:8493/logs/pull-11873-20190520-171414-f0f5d623-verify-rhel-8-0-distropkg/log.html#245) is persistent.

**Comment by KKoukiou** [comment]
> Yep, the [rhel-8-0 failure](https://209.132.184.41:8493/logs/pull-11873-20190520-171414-f0f5d623-verify-rhel-8-0-distropkg/log.html#245) is persistent.

Ah it's the Promise.reject which when dispatched it's not supported by the old cockpit on the distropkg image. Changed to use cockpit.reject instead.

File: test/verify/check-machines-dbus (hunk 1 above)
Comment (KKoukiou) [reply test/verify/check-machines-dbus]: Done

**Pull request merged**